src/components/SortableArea.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  closestCenter,
+  DndContext,
+  DragEndEvent,
+  KeyboardSensor,
+  PointerSensor,
+  UniqueIdentifier,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  sortableKeyboardCoordinates,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import { CSS } from "@dnd-kit/utilities";
+import { useSortable } from "@dnd-kit/sortable";
+import { Drag } from "./icons";
+
+type Props<T> = {
+  children: (item: T, index: number) => React.ReactNode;
+  setItems: (items: T[]) => void;
+  items: T[];
+  disableDragByIndex?: number;
+};
+
+function ItemWrapper({
+  children,
+  id,
+  disabledDrag,
+}: {
+  children: React.ReactNode;
+  id: UniqueIdentifier;
+  disabledDrag: boolean;
```

Comment:
```suggestion
  isDraggable: boolean;
```

or 

```suggestion
  hasDragDisabled: boolean;
```

if you don't want to invert the logic